src/jdk.compiler/share/classes/com/sun/tools/javac/parser/DocCommentParser.java

```diff
@@ -145,83 +199,186 @@ public DCDocComment parse() {
-        return blockContent(Phase.BODY);
+        // The whitespace after a tag name is typically problematic:
+        // should it be included in the content or not?
+        // The problem is made worse by Markdown, in which leading
+        // whitespace in Markdown content may be significant.
+        // While generally in traditional comments, whitespace
+        // after a tag name is skipped, to allow for Markdown we just
+        // skip horizontal whitespace.
+        while (isHorizontalWhitespace(ch) && bp < buflen) {
+            nextChar();
+        }
+        return content(Phase.BODY);
     }
 
     /**
-     * Read block content, consisting of text, html and inline tags.
-     * Terminated by the end of input, or the beginning of the next block tag:
-     * that is, @ as the first non-whitespace character on a line.
+     * Reads "rich text" content, consisting of text, html and inline tags,
+     * according to the given {@code phase}.
+     *
+     * Inline tags are only recognized in {@code BODY} and {@code INLINE}
+     * phases, and not in {@code PREAMBLE} and {@code POSTAMBLE} phases.
+     *
+     * The end of the content is dependent on the phase:
+     *
+     * <ul>
+     * <li>{@code PREAMBLE}: the appearance of {@code <body>} (or {@code <main>}),
+     *      as determined by {@link #isEndPreamble()}
+     * <li>{@code BODY}: the beginning of a block tag, or when reading from
+     *      an HTML file, the appearance of {@code </main>} (or {@code </body>},
+     *      as determined by {@link #isEndBody()}
+     * <li>{@code INLINE}: '}', after skipping any matching {@code { }}
+     * <li>{@code PREAMBLE}: end of file
+     * </ul>
      */
-    @SuppressWarnings("fallthrough")
-    protected List<DCTree> blockContent(Phase phase) {
+    protected List<DCTree> content(Phase phase) {
         ListBuffer<DCTree> trees = new ListBuffer<>();
         textStart = -1;
 
+        int depth = 1;                  // only used when phase is INLINE
+        int pos = bp;                   // only used when phase is INLINE
+        LineKind lineKind = textKind == DocTree.Kind.MARKDOWN ? peekLineKind() : null;
+
         loop:
         while (bp < buflen) {
             switch (ch) {
-                case '\n': case '\r': case '\f':
-                case ' ': case '\t':
+                case '\n', '\r' -> {
                     nextChar();
-                    break;
+                    if (textKind == DocTree.Kind.MARKDOWN) {
+                        int indent = readIndent();
+                        // in the following, the evaluation of INDENTED_CODE_BLOCK is
+                        // inductively a sequence of indented lines following any
+                        // line that is not OTHER
+                        lineKind = (ch == '\n' || ch == '\r') ? LineKind.BLANK
+                                : (indent <= 3) ? peekLineKind()
+                                : lineKind != LineKind.OTHER ? LineKind.INDENTED_CODE_BLOCK
```

Comment:
Doesn't this cause false positives for indented code blocks? In my understanding, indented lines in a list context and directly following a blockquote are not interpreted as code blocks, but as part of the list or blockquote. So my guess would be that `not OTHER` should be something like `BLANK or INDENTED_CODE_BLOCK or HEADER`, and that still leaves the problem of a [blank line in a list context](https://spec.commonmark.org/0.30/#example-108).

Reply:
Sigh. I remember when we all hoped that we wouldn't need to go in the weeds of Markdown parsing, but here we are. Hannes is right. Here are a coupe of problematic examples:

 -     /// List
       ///
       ///   - item
       ///
       ///     @param

 -     /// > Quote
       /// >
       /// >     {@link java.lang.Object}

Reply:
@hns @pavelrappo noted; working on it ...